dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/service/DBStoreEMBuilder.java

```diff
@@ -399,6 +380,93 @@ public DBStoreEMBuilder(DataProvider provider,
             }
         }
 
+        /*
+         * Note: When creating a persistence unit, managed classes (such as entities) are declared in an
+         * all or nothing fashion. Therefore, if we create a persistence unit with a list of entities
+         * we are also required to provide a list of converters, otherwise the persistence provider
+         * will not use them. Ideally, our internal persistence service unit would have a method to
+         * include converter classes alongside entity classes, but the persistence provider API lacks
+         * such function so the converters need to be put into the generated orm.xml file.
+         */
+        Set<Class<?>> converterTypes = new HashSet<>();
+
+        Queue<Class<?>> annotatedEmbeddedClassQueue = new LinkedList<>();
+        Set<Class<?>> embeddedClassesProcessed = new HashSet<>();
+
+        // Discover entities that are indirectly referenced via OneToOne, ManyToMany,
+        // and so forth. Also discover AttributeConverters that are referenced from
+        // classes, fields, and methods.
```

Comment:
praise: wow complex setup here but a very efficient way of iterating through all the possible places we could find an embedd[ed/able], entity, or converter.